### ISSUE TITLE
Separate env vars for OPENAI_API_KEY so openai can be used for expert and openai-compatible for main model

### ISSUE THREAD

**Issue by ai-christianson** [ai-christianson/RA.Aid#26]
If we are using an openai-compatible endpoint for the main agent model, we currently cannot use openai for the expert model because both are trying to use the `OPENAI_API_KEY` env var.

A possible solution is to use `OPENAI_COMPATIBLE_API_KEY` as the env var for openai-compatible providers.

**Comment by leonj1** [comment]
How about a dedicated EXPERT_OPENAI_API_KEY ?

**Comment by ai-christianson** [comment]
Good point :smile:. We already have that capability in there. It's a bit confusing from a UX perspective in that it falls back, unless you are using openai-compatible for the main model, then you don't want fallback if you're actually using openai.
